model-engine/model_engine_server/domain/use_cases/llm_fine_tuning_use_cases.py

```diff
@@ -48,11 +48,20 @@ def ensure_model_name_is_valid_k8s_label(model_name: str):
 
 def read_csv_headers(file_location: str):
     """
-    Read the headers of a csv file. Assumes the file exists and is valid.
+    Read the headers of a csv file. Raises a FileNotFoundError if the file does not exist. Raises a ValueError if csv is malformed.
     """
-    with smart_open.open(file_location, transport_params=dict(buffer_size=1024)) as file:
-        csv_reader = csv.DictReader(file)
-        return csv_reader.fieldnames
+    try:
+        with smart_open.open(file_location, transport_params=dict(buffer_size=1024)) as file:
+            csv_reader = csv.DictReader(file)
+            return csv_reader.fieldnames
+    except FileNotFoundError:
```

Comment:
two ideas:
- move try/except to `CreateFineTuneV1UseCase.execute()` to control exception handling in the main fn and distinguish errors between training/validation file more clearly
- create another "family" of validation helpers that raise all exceptions and move all the exception handling logic from `CreateFineTuneV1UseCase.execute` to these helper fns

Reply:
Makes sense to move things into `CreateFineTuneV1UseCase.execute()`, I'll do that!

Reply:
+1 on moving the try/except here to the use case. Otherwise, if this helper needs to know about the use case, that's not ideal from a least privilege perspective. This helper should just know about reading files and how that can fail.

Reply:
> create another "family" of validation helpers that raise all exceptions and move all the exception handling logic from CreateFineTuneV1UseCase.execute to these helper fns

This can work too, though would just ask that any helpers be localized as much as possible. In other words, if we can continue to keep them in the use case file, instead of a global helpers/common/util file.